app/routes/task_routes.py

```diff
@@ -0,0 +1,108 @@
+import datetime, requests, os
+from app import db 
+from app.models.task import Task 
+from flask import Blueprint, make_response,request, jsonify, abort
+
+task_bp = Blueprint("task", __name__, url_prefix="/tasks")
+
+@task_bp.route("", methods=["POST"])
+def create_task(): 
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body: 
+        return make_response({"details" : "Invalid data"}, 400)
+
+    new_task = Task(
+        title=request_body["title"],
+        description=request_body["description"]
+    )
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks(): 
+    response = []
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc": 
+        all_tasks = Task.query.order_by(Task.title.asc()) 
+    elif sort_query == "desc":
+        all_tasks = Task.query.order_by(Task.title.desc()) 
+    else: 
+        all_tasks = Task.query.all()
+
+    for task in all_tasks: 
+        response.append(task.to_dict())
+    
+    return jsonify(response), 200
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id): 
+    task = validate_item(Task, task_id)
+    return {"task": task.to_dict()}, 200 
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id): 
+    task = validate_item(Task, task_id)
+
+    request_data = request.get_json()
+
+    task.title = request_data["title"]
+    task.description = request_data["description"]
+
+    db.session.commit()
+
+    return {"task": task.to_dict()}, 200
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def update_task_as_complete(task_id): 
+    task = validate_item(Task, task_id)
+
+    task.completed_at = datetime.datetime.now(tz=None)
```

Comment:
```suggestion
    task.completed_at = datetime.datetime.now()
```
default value for `tz` is `None` so you don't need to pass it in as an argument.